src/hotspot/share/gc/g1/g1CollectedHeap.hpp

```diff
@@ -782,7 +782,10 @@ class G1CollectedHeap : public CollectedHeap {
   void calculate_collection_set(G1EvacuationInfo& evacuation_info, double target_pause_time_ms);
 
   // Actually do the work of evacuating the parts of the collection set.
-  void evacuate_initial_collection_set(G1ParScanThreadStateSet* per_thread_states);
+  // The may_do_optional_evacuation flag indicates whether some or more optional evacuation
+  // steps will follow.
+  void evacuate_initial_collection_set(G1ParScanThreadStateSet* per_thread_states,
```

Comment:
The PR description provides a very good motivation for this patch, and I think such motivation should be reprinted here to explain **why** we are interested in tracking if there is an optional evac.

Reply:
I added a little bit of motivation. I am planning to write up more about the card scanning and the various motivations in a better place. I hope the text that is now there is sufficient.

Thanks for your review.

Reply:
> I hope the text that is now there is sufficient.

Sure; thank you.